packages/firestore/test/unit/specs/spec_test_runner.ts

```diff
@@ -1559,6 +1598,11 @@ export type SpecWriteFailure = {
   expectUserCallback: boolean;
 };
 
+export type SpecDrainQueue = {
+  /** Whether the drain is expected to generate a user callback. */
```

Comment:
As mentioned elsewhere I wonder if we can/should allow for multiple callbacks.

Reply:
This is now gone and we do allow for multiple callbacks.